src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -179,28 +269,37 @@ class CVulkanPhysicalDevice final : public IPhysicalDevice
             qci.flags = static_cast<VkDeviceQueueCreateFlags>(qparams.flags);
             qci.pQueuePriorities = qparams.priorities;
         }
-        createInfo.pQueueCreateInfos = queueCreateInfos.data();
-                
+        vk_createInfo.pQueueCreateInfos = queueCreateInfos.data();
+
+        vk_createInfo.enabledLayerCount = 1u; // deprecated and ignored param
+        const char* validationLayerName[] = { "VK_LAYER_KHRONOS_validation" };
+        vk_createInfo.ppEnabledLayerNames = validationLayerName; // deprecated and ignored param
+
         // Todo(achal): Need to get this from the user based on if its a headless compute or some presentation worthy workload
         const uint32_t deviceExtensionCount = 1u;
         const char* deviceExtensions[] = { VK_KHR_SWAPCHAIN_EXTENSION_NAME };
                 
-        createInfo.enabledExtensionCount = deviceExtensionCount;
-        createInfo.ppEnabledExtensionNames = deviceExtensions;
-                
+        vk_createInfo.enabledExtensionCount = deviceExtensionCount;
+        vk_createInfo.ppEnabledExtensionNames = deviceExtensions;
+        
         // Todo(achal): Need to get this from the user, which features they want
-        VkPhysicalDeviceFeatures deviceFeatures = {};
-                
-        createInfo.pEnabledFeatures = &deviceFeatures;
-                
-        VkDevice vkdev = VK_NULL_HANDLE;
-        assert(vkCreateDevice(m_vkphysdev, &createInfo, nullptr, &vkdev) == VK_SUCCESS);
+        VkPhysicalDeviceFeatures vk_deviceFeatures = {};        
+        vk_createInfo.pEnabledFeatures = &vk_deviceFeatures;        
                 
-        return core::make_smart_refctd_ptr<CVKLogicalDevice>(vkdev, params, core::smart_refctd_ptr(m_system), core::smart_refctd_ptr(m_GLSLCompiler));
+        VkDevice vk_device = VK_NULL_HANDLE;
+        if (vkCreateDevice(m_vkPhysicalDevice, &vk_createInfo, nullptr, &vk_device) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVKLogicalDevice>(this, vk_device, params,
```

Comment:
remember, make the constructor protected, and use the two step two line smartpointer creation trick